crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1238 @@
+    }
+    gmap.add_instruction(UGate(theta, phi, lam), u_props)
+    cx_props = {
+        (0,1): InstructionProperties(duration=5.23e-7, error=0.00098115),
+        (1,0): InstructionProperties(duration=4.52e-7, error=0.00132115),
+    }
+    gmap.add_instruction(CXGate(), cx_props)
+
+Each instruction in the ``Target`` is indexed by a unique string name that uniquely
+identifies that instance of an :class:`~qiskit.circuit.Instruction` object in
+the Target. There is a 1:1 mapping between a name and an
+:class:`~qiskit.circuit.Instruction` instance in the target and each name must
+be unique. By default, the name is the :attr:`~qiskit.circuit.Instruction.name`
+attribute of the instruction, but can be set to anything. This lets a single
+target have multiple instances of the same instruction class with different
+parameters. For example, if a backend target has two instances of an
+:class:`~qiskit.circuit.library.RXGate` one is parameterized over any theta
+while the other is tuned up for a theta of pi/6 you can add these by doing something
+like::
+
+    import math
+
+    from qiskit.transpiler import Target, InstructionProperties
+    from qiskit.circuit.library import RXGate
+    from qiskit.circuit import Parameter
+
+    target = Target()
+    theta = Parameter('theta')
+    rx_props = {
+        (0,): InstructionProperties(duration=5.23e-8, error=0.00038115),
+    }
+    target.add_instruction(RXGate(theta), rx_props)
+    rx_30_props = {
+        (0,): InstructionProperties(duration=1.74e-6, error=.00012)
+    }
+    target.add_instruction(RXGate(math.pi / 6), rx_30_props, name='rx_30')
+
+Then in the ``target`` object accessing by ``rx_30`` will get the fixed
+angle :class:`~qiskit.circuit.library.RXGate` while ``rx`` will get the
+parameterized :class:`~qiskit.circuit.library.RXGate`.
+
+.. note::
+
+    This class assumes that qubit indices start at 0 and are a contiguous
+    set if you want a submapping the bits will need to be reindexed in
+    a new``Target`` object.
+
+.. note::
+
+    This class only supports additions of gates, qargs, and qubits.
+    If you need to remove one of these the best option is to iterate over
+    an existing object and create a new subset (or use one of the methods
+    to do this). The object internally caches different views and these
+    would potentially be invalidated by removals.
+ */
+#[pyclass(mapping, subclass, module = "qiskit._accelerate.target")]
+#[derive(Clone, Debug)]
+pub struct BaseTarget {
+    #[pyo3(get, set)]
+    pub description: Option<String>,
+    #[pyo3(get)]
+    pub num_qubits: Option<usize>,
+    #[pyo3(get, set)]
+    pub dt: Option<f64>,
+    #[pyo3(get, set)]
+    pub granularity: i32,
+    #[pyo3(get, set)]
+    pub min_length: usize,
+    #[pyo3(get, set)]
+    pub pulse_alignment: i32,
+    #[pyo3(get, set)]
+    pub acquire_alignment: i32,
+    #[pyo3(get, set)]
+    pub qubit_properties: Option<Vec<PyObject>>,
```

Comment:
You left this as a `PyObject` to avoid having to define a struct for this in Rust? I think this is fine for a temporary thing but we'll want to represent this in Rust eventually (it's not critical for this PR). Can you add a `TODO` comment here to track a follow on work item to move this to rust.